events/syscall/fileless_execution_via_memfd_create.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package syscall
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/falcosecurity/event-generator/events"
+)
+
+var _ = events.Register(FilelessExecutionViaMemfdCreate)
+
+func FilelessExecutionViaMemfdCreate(h events.Helper) error {
+	sourceCode := `package main
+	import "fmt"
+	
+	func main() {
+		fmt.Println("Hello, world! This is a fileless execution example.")
+	}`
+
+	binaryPath := "/tmp/program"
+	err := os.WriteFile(binaryPath+".go", []byte(sourceCode), 0644)
+	if err != nil {
+		h.Log().WithError(err).Error("failed to write source code to file")
+		return err
+	}
+	defer os.Remove(binaryPath + ".go")
+
+	// Compile the Go source code into an executable binary
+	compileCmd := exec.Command("go", "build", "-o", binaryPath, binaryPath+".go")
```

Comment:
This requires Go to be installed as a pre-requisite, doesn't it?

 If so, this action won't work on many systems, unfortunately. Is there any alternative approach?

Reply:
I thought like we are executing a .go file for running this event so then go should exist in system right to run this event?

Reply:
I believe rule triggers just if one tries to execute a binary (regardless if the binary works or not). If that's true, I don't think we need to build the executable here or depend on Go.

Have you tried to use an empty file instead of a real binary? :thinking:

Reply:
@leogr Yes when I use an binary which is not executable then rule wont triggers!
Bcz as one condition in the rule is `proc.is_exe_from_memfd = true` which is defined like below in [doumentaion](https://falco.org/docs/reference/rules/supported-fields/)

> proc.is_exe_from_memfd  is 'true' if the executable file of the current process is an anonymous file created using memfd_create() and is being executed by referencing its file descriptor (fd). This type of file exists only in memory and not on disk. Relevant to detect malicious in-memory code injection. Requires kernel version greater or equal to 3.17.0.

So it means only a process is launched if binary is executable

Reply:
Ok got it.
A more complex and clean solution may be to use the `event-generator` binary itself (which is already available for sure).

This would require some kind of [helper](https://github.com/falcosecurity/event-generator/blob/main/pkg/runner/helper.go) to return the path of the current executable. 

Would it work for you? Wdyt?

cc @loresuso @FedeDP

Reply:
Ok let me try and update you with the results!

Reply:
Yes totally agreed! This is how would implement fileless execution too :)